### PR TITLE
Rest framework

### PR DESCRIPTION
**Pull request by zaneadix**
This is definitely not done but theres a pretty solid start. I really just wanted to see if you'd like to take a look at whats been done so far and whether or not things should go in another direction.

In the v2 views I've got some abstracted behavior for allowing query by name or id so

api/v2/ability/1 and api/v2/ability/stench

will serve the same thing. Also make a request to the base resource will return a very limited list of resource URIs matched to names rather than complete resources. This is something Im not 100% on because I'm not super informed on best practices for API structure but thought it made sense.

api/v2/ability returns
```
{
count: 248,
next: "http://localhost:8000/api/v2/ability/?limit=20&offset=20",
previous: null,
results: [
{
url: "http://localhost:8000/api/v2/ability/1/",
name: "stench",
is_main_series: true,
generation: "http://localhost:8000/api/v2/generation/3/"
},
{
url: "http://localhost:8000/api/v2/ability/2/",
name: "drizzle",
is_main_series: true,
generation: "http://localhost:8000/api/v2/generation/3/"
},
{
url: "http://localhost:8000/api/v2/ability/3/",
name: "speed-boost",
is_main_series: true,
generation: "http://localhost:8000/api/v2/generation/3/"
},
```
and api/v2/ability/1 or api/v2/ability/stench returns

```
{
id: 1,
name: "stench",
is_main_series: true,
generation: {
name: "generation-iii",
url: "http://localhost:8000/api/v2/generation/3/"
},
names: [
{
name: "Stench",
language: {
name: "en",
url: "http://localhost:8000/api/v2/language/9/"
}
},
{
name: "Tanfo",
language: {
name: "it",
url: "http://localhost:8000/api/v2/language/8/"
}
},
{
name: "Hedor",
language: {
name: "es",
url: "http://localhost:8000/api/v2/language/7/"
}
},
{
name: "Duftnote",
language: {
name: "de",
url: "http://localhost:8000/api/v2/language/6/"
}
},
{
name: "Puanteur",
language: {
name: "fr",
url: "http://localhost:8000/api/v2/language/5/"
}
},
{
name: "ì•…ì·¨",
language: {
name: "ko",
url: "http://localhost:8000/api/v2/language/3/"
}
},
{
name: "ã‚ãã—ã‚…ã†",
language: {
name: "ja",
url: "http://localhost:8000/api/v2/language/1/"
}
}
],
descriptions: [
{
effect: "This PokÃ©mon's damaging moves have a 10% chance to make the target [flinch]{mechanic:flinch} with each hit if they do not already cause flinching as a secondary effect. This ability does not stack with a held item. Overworld: The wild encounter rate is halved while this PokÃ©mon is first in the party.",
short_effect: "Has a 10% chance of making target PokÃ©mon [flinch]{mechanic:flinch} with each hit.",
language: {
name: "en",
url: "http://localhost:8000/api/v2/language/9/"
}
}
],
flavor_text_entries: [
{
text: "Helps repel wild POKÃ©MON.",
version_group: {
name: "ruby-sapphire",
url: "http://localhost:8000/api/v2/version-group/5/"
},
language: {
name: "en",
url: "http://localhost:8000/api/v2/language/9/"
}
},
{
text: "Helps repel wild POKÃ©MON.",
version_group: {
name: "emerald",
url: "http://localhost:8000/api/v2/version-group/6/"
},
language: {
name: "en",
url: "http://localhost:8000/api/v2/language/9/"
}
},
{
text: "Helps repel wild POKÃ©MON.",
version_group: {
name: "firered-leafgreen",
url: "http://localhost:8000/api/v2/version-group/7/"
},
language: {
name: "en",
url: "http://localhost:8000/api/v2/language/9/"
}
},
{
text: "The stench helps keep wild PokÃ©mon away.",
version_group: {
name: "diamond-pearl",
url: "http://localhost:8000/api/v2/version-group/8/"
},
language: {
name: "en",
url: "http://localhost:8000/api/v2/language/9/"
}
},
{
text: "The stench helps keep wild PokÃ©mon away.",
version_group: {
name: "platinum",
url: "http://localhost:8000/api/v2/version-group/9/"
},
language: {
name: "en",
url: "http://localhost:8000/api/v2/language/9/"
}
},
{
text: "The stench helps keep wild PokÃ©mon away.",
version_group: {
name: "heartgold-soulsilver",
url: "http://localhost:8000/api/v2/version-group/10/"
},
language: {
name: "en",
url: "http://localhost:8000/api/v2/language/9/"
}
},
{
text: "La puanteur peut effrayer lâ€™adversaire.",
version_group: {
name: "black-white",
url: "http://localhost:8000/api/v2/version-group/11/"
},
language: {
name: "fr",
url: "http://localhost:8000/api/v2/language/5/"
}
},
{
text: "The stench may cause the target to flinch.",
version_group: {
name: "black-white",
url: "http://localhost:8000/api/v2/version-group/11/"
},
language: {
name: "en",
url: "http://localhost:8000/api/v2/language/9/"
}
},
{
text: "The stench may cause the target to flinch.",
version_group: {
name: "black-2-white-2",
url: "http://localhost:8000/api/v2/version-group/14/"
},
language: {
name: "en",
url: "http://localhost:8000/api/v2/language/9/"
}
},
{
text: "ãã•ãã¦ã€€ã‚ã„ã¦ãŒ ã²ã‚‹ã‚€ã€€ã“ã¨ãŒã‚ã‚‹ã€‚",
version_group: {
name: "x-y",
url: "http://localhost:8000/api/v2/version-group/15/"
},
language: {
name: "ja",
url: "http://localhost:8000/api/v2/language/1/"
}
},
{
text: "ì•…ì·¨ ë•Œë¬¸ì— ìƒëŒ€ê°€ í’€ì£½ì„ ë•Œê°€ ìžˆë‹¤.",
version_group: {
name: "x-y",
url: "http://localhost:8000/api/v2/version-group/15/"
},
language: {
name: "ko",
url: "http://localhost:8000/api/v2/language/3/"
}
},
{
text: "La puanteur peut effrayer lâ€™adversaire.",
version_group: {
name: "x-y",
url: "http://localhost:8000/api/v2/version-group/15/"
},
language: {
name: "fr",
url: "http://localhost:8000/api/v2/language/5/"
}
},
{
text: "LÃ¤sst den Gegner durch Gestank zurÃ¼ckschrecken.",
version_group: {
name: "x-y",
url: "http://localhost:8000/api/v2/version-group/15/"
},
language: {
name: "de",
url: "http://localhost:8000/api/v2/language/6/"
}
},
{
text: "Es posible que el rival retroceda por el mal olor.",
version_group: {
name: "x-y",
url: "http://localhost:8000/api/v2/version-group/15/"
},
language: {
name: "es",
url: "http://localhost:8000/api/v2/language/7/"
}
},
{
text: "A volte il cattivo odore fa tentennare i nemici.",
version_group: {
name: "x-y",
url: "http://localhost:8000/api/v2/version-group/15/"
},
language: {
name: "it",
url: "http://localhost:8000/api/v2/language/8/"
}
},
{
text: "The stench may cause the target to flinch.",
version_group: {
name: "x-y",
url: "http://localhost:8000/api/v2/version-group/15/"
},
language: {
name: "en",
url: "http://localhost:8000/api/v2/language/9/"
}
},
{
text: "è‡­ãã¦ã€€ç›¸æ‰‹ãŒ ã²ã‚‹ã‚€ã€€ã“ã¨ãŒã‚ã‚‹ã€‚",
version_group: {
name: "x-y",
url: "http://localhost:8000/api/v2/version-group/15/"
},
language: {
name: "ja-kanji",
url: "http://localhost:8000/api/v2/language/11/"
}
}
]
}
```

Sorry for the ugly formatting. Im in kind of a rush and just wanted to show you whats going on. I've included all names and descriptions in the detail views along with their associated languages.

If you have a chance to get it up and running thatd be awesome and would give you a much better perspective on what Ive done.

Currently everything being served:
Languages
Regions
Generations
Versions
Abilities
Types

and others just not as fleshed out.

### CODE REVIEW

File: .bash_profile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (phalt) [inline .bash_profile]: Has this been committed by mistake?

File: config/settings.py [hunk 2]
```diff
@@ -83,7 +89,7 @@
     'default': {
         'ENGINE': 'django.db.backends.postgresql_psycopg2',
         'NAME': 'pokeapi_co_db',
-        'USER': 'root',
+        'USER': 'zane',
```
Comment (phalt) [inline config/settings.py]: This needs to be kept as "root"

**Comment by zaneadix** [comment]
Yeah sorry. I need to fix those things. bash_profile definitely shouldnt be there and for some reason root access has been weird on my database so I've been working against a different user.

**Comment by zaneadix** [comment]
Just wanna double check with you and make sure you think we'ed be safe to merge this work in for now. Just committed some more work. Below are all urls that are serving data.

api/v2/ability
api/v2/berry
api/v2/berry-firmness
api/v2/egg-group
api/v2/evolution-chain
api/v2/evolution-trigger
api/v2/generation
api/v2/growth-rate
api/v2/item
api/v2/item-category
api/v2/item-attribute
api/v2/item-fling-effect
api/v2/item-pocket
api/v2/language
api/v2/location
api/v2/move
api/v2/move-ailment
api/v2/move-category
api/v2/move-damage-class
api/v2/move-learn-method
api/v2/move-target
api/v2/nature
api/v2/pokedex
api/v2/pokemon
api/v2/pokemon-habitat
api/v2/pokemon-shape
api/v2/pokemon-species
api/v2/pokemon-color
api/v2/pokemon-form
api/v2/region
api/v2/stat
api/v2/type
api/v2/version
api/v2/version-group

**Comment by phalt** [comment]
Wow nice. Let's get it merged. Let's get some tests in next that test each endpoint and makes sure the responses are what we expect?

**Comment by zaneadix** [comment]
OK cool. I'll get focused on tests next.

**Pull request merged**